### PR TITLE
PATCH facility default athena mapping id

### PR DESCRIPTION
**Pull request by thomasyopes**
Ref: #1040

### Description

- updating the default external ID to be practice specific -- this allows us to do facility per practice and gets rid of the constraint conflict of multiple `default / athenahealth` entries.

### Release Plan

- :warning: Points to `master`
- [ ] Update existing facility mappings for athena
- [ ] Merge this

### CODE REVIEW

File: packages/api/src/command/mapping/facility.ts [hunk 1]
```diff
@@ -53,24 +54,53 @@ export async function getFacilityMappingOrFail({
...
   });
-  if (!existing) {
-    throw new NotFoundError("Entry not found", undefined, { cxId, externalId, source });
+  if (!mapping) {
+    throw new NotFoundError("FacilityMappingModel not found", undefined, { cxId, id });
   }
+  return mapping;
+}
+
+export async function updateFacilityMapping({
```
[the remark this answers is not in the document]
Comment (thomasyopes) [reply packages/api/src/command/mapping/facility.ts:90]: ditto above

File: packages/api/src/routes/internal.ts [hunk 2]
```diff
@@ -350,24 +352,46 @@ router.get(
...
  */
-router.delete(
-  "/cx-mapping",
+router.put(
+  "/cx-mapping/:id",
   requestLogger,
   asyncHandler(async (req: Request, res: Response) => {
-    const source = getFromQueryOrFail("source", req);
+    const cxId = getUUIDFrom("query", req, "cxId").orFail();
+    const id = getFrom("params").orFail("id", req);
```
[the remark this answers is not in the document]
Comment (thomasyopes) [reply packages/api/src/routes/internal.ts:367]: id is the row id, not external id.

File: packages/api/src/command/mapping/cx.ts [hunk 3]
```diff
@@ -57,23 +51,55 @@ export async function getCxMappingOrFail({
   return mapping;
 }
 
-export async function getCxMappingsForCustomer(where: {
+export async function getCxMappingsBySource(where: { source: CxSources }): Promise<CxMapping[]> {
+  const mappings = await CxMappingModel.findAll({ where });
```
[the remark this answers is not in the document]
Comment (thomasyopes) [reply packages/api/src/command/mapping/cx.ts]: Done.

File: packages/api/src/command/mapping/cx.ts [hunk 4]
```diff
@@ -57,23 +51,55 @@ export async function getCxMappingOrFail({
...
+    cxId,
+    id,
+  });
+  if (!mapping) {
+    throw new NotFoundError("CxMappingModel not found", undefined, { cxId, id });
   }
+  return mapping;
+}
+
+export async function updateCxMapping({
```
[the remark this answers is not in the document]
Comment (thomasyopes) [reply packages/api/src/command/mapping/cx.ts]: Done.

File: packages/api/src/command/mapping/facility.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (thomasyopes) [reply packages/api/src/command/mapping/facility.ts]: Done.

File: packages/api/src/routes/internal.ts [hunk 5]
```diff
@@ -354,19 +354,19 @@ router.get(
...
  *
- * Update cx mapping
+ * Update cx mapping external ID
  *
  * @param req.query.cxId - The cutomer's ID.
  * @param req.query.externalId - Mapped external ID.
  */
 router.put(
-  "/cx-mapping/:id",
+  "/cx-mapping/:id/external-id",
```
Comment (leite08) [inline packages/api/src/routes/internal.ts:363]: 😍

File: packages/api/src/command/mapping/facility.ts [hunk 6]
```diff
@@ -53,24 +54,58 @@ export async function getFacilityMappingOrFail({
...
+async function getFacilityMappingModelByIdOrFail({
+  cxId,
+  id,
+}: FacilityMappingLookupByIdParams): Promise<FacilityMappingModel> {
+  const mapping = await getFacilityMappingModelById({
+    cxId,
+    id,
+  });
+  if (!mapping) {
+    throw new NotFoundError("FacilityMappingModel not found", undefined, { cxId, id });
```
Comment (leite08) [inline packages/api/src/command/mapping/facility.ts:90]: nit: when I read this message I think the file/class is not available... 😄

What do you think we reword it to `Facility mapping not found`?

File: packages/api/src/command/mapping/facility.ts [hunk 7]
```diff
@@ -53,24 +54,58 @@ export async function getFacilityMappingOrFail({
...
+}
+
+export async function setExternalIdOnFacilityMapping({
+  cxId,
+  id,
+  externalId,
+}: FacilityMappingLookupByIdParams & { externalId: string }): Promise<FacilityMapping> {
+  const mapping = await getFacilityMappingModelByIdOrFail({ cxId, id });
+  await mapping.update({ externalId });
+  return mapping.dataValues;
```
Comment (leite08) [inline packages/api/src/command/mapping/facility.ts:102]: I think we're returning the original, unmodified data here. We should prob return the updated version, right?

Comment (thomasyopes) [reply packages/api/src/command/mapping/facility.ts:102]: good catch, definitely

Comment (thomasyopes) [reply packages/api/src/command/mapping/facility.ts]: done.

File: packages/api/src/command/mapping/facility.ts (hunk 6 above)
Comment (thomasyopes) [reply packages/api/src/command/mapping/facility.ts]: done.
